### PR TITLE
Fix/jwt auth password reset

### PR DESCRIPTION
**Pull request by klesogor**
created api/v1/auth/cahnge method to reset current password

### CODE REVIEW

File: app/Http/Controllers/Api/Auth/AuthController.php [hunk 1]
```diff
@@ -112,4 +115,16 @@ public function recoverPassword(RecoverPasswordHttpRequest $httpRequest, Recover
             return $this->errorResponse(AuthPresenter::presentError($exception), 400);
         }
     }
+
+    public function cahngePassword(ChangePasswordHttpRequest $httpRequest,ChangePasswordAction $action)
```
Comment (nemoipaha) [inline app/Http/Controllers/Api/Auth/AuthController.php]: cahngePassword fix name

File: routes/api.php [hunk 2]
```diff
@@ -32,6 +32,8 @@
             Route::post('/refresh', 'AuthController@refresh');
 
             Route::get('/me', 'AuthController@me');
+
+            Route::post('/change', 'AuthController@cahngePassword');
```
Comment (nemoipaha) [inline routes/api.php]: name reset-password is better )

File: tests/Feature/Api/Auth/ResetPasswordTest.php [hunk 3]
```diff
@@ -6,6 +6,7 @@
 use Hedonist\Events\Auth\PasswordResetedEvent;
 use Illuminate\Foundation\Testing\RefreshDatabase;
 use Illuminate\Support\Facades\Event;
+use Illuminate\Support\Facades\Hash;
 use Tests\TestCase;
 
 class ResetPasswordTest extends TestCase
```
Comment (nemoipaha) [inline tests/Feature/Api/Auth/ResetPasswordTest.php]: extend ApiTestCase

File: tests/Feature/Api/Auth/ResetPasswordTest.php [hunk 4]
```diff
@@ -74,4 +75,51 @@ public function test_incorrect_token()
 
         $response->assertStatus(400);
     }
+
+    public function test_change_password_success()
```
Comment (nemoipaha) [inline tests/Feature/Api/Auth/ResetPasswordTest.php]: We have actingAsToken method in ApiTestCase

Comment (klesogor) [reply tests/Feature/Api/Auth/ResetPasswordTest.php]: used authenticate method instead

File: app/Http/Controllers/Api/Auth/AuthController.php [hunk 5]
```diff
@@ -112,4 +115,16 @@ public function recoverPassword(RecoverPasswordHttpRequest $httpRequest, Recover
...
         }
     }
+
+    public function changePassword(ChangePasswordHttpRequest $httpRequest,ChangePasswordAction $action)
+    {
+        try {
+            $changeRequest = new ChangePasswordRequest($httpRequest->old_password,$httpRequest->new_password);
+            $action->execute($changeRequest);
+
+            return $this->successResponse([]);
```
Comment (nemoipaha) [inline app/Http/Controllers/Api/Auth/AuthController.php]: Better to use $this->emptyResponse() here.

**Comment by klesogor** [comment]
Fixed remarks. CI will fail because of PlaceController issue

**Pull request merged**